Mathlib/Data/Set/Subset.lean

```diff
@@ -0,0 +1,260 @@
+/-
+Copyright (c) 2024 Miguel Marco. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Miguel Marco
+-/
+import Mathlib.Data.Set.Function
+import Mathlib.Data.Set.Functor
+import Mathlib.Lean.Expr.ExtraRecognizers
+
+/-!
+# Sets in subtypes
+
+This file defines notation for sets in a type pulled down to sets in a subtype, and sets in
+a subtype, lifted up to sets in the ambient type.
+
+It also provides some related lemmas for convenience.
+
+## Notation
+
+Let `α` be a `Type`, `A B : Set α` two sets in `α`, and `C : Set ↑A` a set in the subtype `↑A`.
+
+- `A ↓∩ B` denotes `(Subtype.val ⁻¹' B : Set A)` (that is, `{x : ↑A | ↑x ∈ B}`).
+- `↑C` denotes `Subtype.val '' C` (that is, `{x : α | ∃ y ∈ C, ↑y = x}`).
+
+To access this notation, type `open Subset`.
```

Comment:
I think this should all be in the `Set` namespace rather than this new `Subset` namespace. Could you change `namespace Subset` to `namespace Set`?

It's defensible putting your notation into a special notation namespace. Let's put `scoped notation3` in the `Set.Notation` namespace. That way, you can do `open Set.Notation` to get this extra notation. I think a convenient syntax you can use for this is `scoped[Set.Notation] notation3 ...` rather than wrapping it in a namespace.

```suggestion
This notation is scoped to the `Set.Notation` namespace.
To enable it, use `open scoped Set.Notation`.
```

Reply:
I didn't find a way to scope the delaborator, so I wrapped it and the notation in a namespace.